Document new build jobs (#11900)
readthedocs/readthedocs.org: docs/user/config-file/v2.rst

.. note::

>    Each key under ``build.jobs`` must be a list of strings
>    (except for ``build.jobs.build`` which is a dictionary, as described below).

Reviewer: ```suggestion
   Each key under ``build.jobs`` must be a list of strings
   except for ``build.jobs.build`` which is a dictionary,
   with keys for each build format. 
```

       python: "3.13"
     jobs:
       create_environment:
         - echo "Preparing environment"
       install:
         - echo "Installing dependencies"
       build:
         html:
           - echo "Building HTML"
           - mkdir -p $READTHEDOCS_OUTPUT/html/
           - echo "Hello world!" > $READTHEDOCS_OUTPUT/html/index.html
         pdf:
           - echo "Building PDF"
           - mkdir -p $READTHEDOCS_OUTPUT/pdf/
           - echo "Hello world!" > $READTHEDOCS_OUTPUT/pdf/index.pdf
         epub:
           - echo "Building ePub"
           - mkdir -p $READTHEDOCS_OUTPUT/epub/
           - echo "Hello world!" > $READTHEDOCS_OUTPUT/epub/index.epub

:Type: ``dict``
:Allowed keys: ``html``, ``pdf``, ``epub``, ``htmlzip``
:Required: ``false``
:Default: ``{}``

.. note::

   If any of the ``pdf``, ``epub``, or ``htmlzip`` steps are overridden,
   they should be included in the :ref:`config-file/v2:formats` list.

.. note::

   For each format, you need to generate their output in the :ref:`$READTHEDOCS_OUTPUT <build-customization:where to put files>` directory.

build.commands
``````````````

Specify a list of commands that Read the Docs will run on the build process.

> When ``build.commands`` is used, none of the :term:`pre-defined build jobs` will be executed.

Reviewer 2: I think we should keep the link to ```(see :doc:`/build-customization` for more details)``` here.
